RELEASING.md

```diff
@@ -0,0 +1,31 @@
+# Releasing
+
+## Release checklist
```

Comment:
I love this addition!

This has a big overlap with https://github.com/haskell/haskell-language-server/blob/master/docs/contributing/releases.md, we should probably merge these documents.

Reply:
Possibly... but the other document is old and I've never looked at it.

My suggestion is to have another contributor do a release while I'm available and ask questions when anything in the checklist is unclear... and then we can add more information as we go.

Reply:
An acceptable merge operation is "discard the old one" :)

Reply:
The old list is at least partially up-to-date and contains a lot of detailed information, I would not just delete it.